[antgyf] iP

antgyf opened the pull request:
# **PHamBot**

> Hi! I'm PHamBot! Happy to be of service to you today!

Want to track you To-Do's, deadlines, and events? Use PHamBot!
- Fast
- Easy to Use
- **Free!**

# How to use it?

1. **Download** it from the github repo [here] (https://github.com/antgyf/ip/tree/master).
2. **Compile and run** the PHamBot java file
3. Let PHamBot handle things for you 👍 

# **What's Done?**
- [x] Implemented working prototype for managing tasks, deadline, and events
- [x] Able to find, delete, mark or unmark items as completed
- [ ] Implemented GUI

## Want to `test` the product? 

Run the following in Gradle
```
./gradlew test
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

antgyf committed:
Week 2 level 0

antgyf committed:
Week2 Level-1

antgyf committed:
Week2 Level 2

antgyf committed:
Week2 Level 3

antgyf committed:
Week2 Level 4

antgyf committed:
Automated testing

antgyf committed:
Week2 Level-5

antgyf committed:
Week2 Level-6 in progress

antgyf committed:
Week2 Level-6 in progress

antgyf committed:
Week2 Level-6 done

antgyf committed:
Level-7 in progress

antgyf committed:
Week3 Level-7

antgyf committed:
Week3 Level-8

antgyf committed:
Week3 A-MoreOOP

antgyf committed:
bugfixing

antgyf committed:
week3 A-Packages

antgyf committed:
forgot to stage data folder

antgyf committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

antgyf committed:
Week3 A-Gradle

antgyf committed:
Week3 A-JUnit

OttoTan commented on src/main/java/PHambot/parser/Parser.java:
```diff
@@ -0,0 +1,149 @@
+        }
+    }
+
+    private int prepIndexedCommand(String arguments) {
```
Should the exception been thrown here instead of passing in -1 to throw the exception further down the line?

OttoTan commented on src/main/java/PHambot/storage/UserData.java:
```diff
@@ -0,0 +1,102 @@
+        this.tasks = tasks;
+    }
+
+    public void saveTasks() {
```
Should the user be made aware if an error occurs?

OttoTan reviewed (commented):
Overall LGTM, code is nice and segmented, with each class serving its distinct purpose 👍

antgyf committed:
docs: Add Javadocs for classes, methods and fields

antgyf committed:
Fix some coding standard violations

antgyf committed:
Add FindCommand class and update Parser class

antgyf committed:
Add and use CheckStyle to detect coding violations

antgyf committed:
Update gradle dependencies

Arren11111 commented on src/main/java/PHambot/storage/UserData.java:
```diff
@@ -0,0 +1,127 @@
+            String task;
+            while ((task = reader.readLine()) != null) {
+                String[] splitTask = task.split("/");
+                if (splitTask[0].equals("T")) {
```
Would be more logically accurate if you used "else if" to run through the situations here

Arren11111 commented on src/main/java/PHambot/utils/Utilities.java:
```diff
@@ -0,0 +1,21 @@
+    /**
+     * A constant string representing a line separator.
+     */
+    public static final String line = "____________________________________________________________\n";
```
constants should be capitalised

Arren11111 reviewed (commented):
Only some minor details and convention errors to fix. Great job!

antgyf committed:
Add GUI using javaFX, update chatbot theme

antgyf committed:
Changes to exception handling to work with new GUI

antgyf committed:
Add assertions through codebase

Inserting assertions at key points in functions helps to ensure correctness of key variables and return values.

This will improve code reliability, ensuring mistakes are caught early on.

antgyf committed:
Parser: Make use of streams for greetings

Made use of streams in handling different possible greetings from the user.

This serves to simplify and streamline the code, removing the need for a chunky loop to check for greetings.

antgyf committed:
Parser: Rewrite code for prep helper functions

There are several prep<Task> methods that are used to return a new task instance.

Rewriting the code to be more readable will prevent confusion in the future, and make future changes easier.

antgyf committed:
Merge pull request #2 from antgyf/branch-A-Streams

Parser: Make use of streams for greetings

antgyf committed:
Merge branch 'master' into branch-A-CodeQuality

antgyf committed:
Merge pull request #3 from antgyf/branch-A-CodeQuality

Branch a code quality

antgyf committed:
Merge branch 'master' into branch-A-Assertions

antgyf committed:
Merge pull request #1 from antgyf/branch-A-Assertions

Add assertions through codebase

antgyf committed:
Bugfixing

antgyf committed:
Add support for optional tags and implement findtag functionality

Tasks currently lack the ability to be categorized with tags, and
there is no way to retrieve tasks based on specific tags.
Introduce a Tag class and modify task creation to include optional
tags.
Implement findtag functionality to return tasks with matching
tags.
This approach allows for better organization and filtering of tasks
by tags, improving usability and flexibility. The use of optional
tags ensures backward compatibility for tasks without tags.

antgyf committed:
Parser class: minor code reformat

Reformat of else statements in parser class,
to be more in line with expected standard.

antgyf committed:
Parser class: bugfix for handling tagging

Addition of Tags created a few unexpected issues.

Make code adjustments to properly account for the new tag arguments.

antgyf committed:
IndexedCommand class: introduce new class to handle index commands

Currently there are a few commands that need indexes as arguments

It would make sense to create a parent IndexedCommand class for shared fields.

antgyf committed:
Command test classes: create more test classes

implement test suites for the Johnson.command package.

Test cases will test most of the foreseeable use cases.

antgyf committed:
UserDataTest class

create test cases for the UserData class.

Additionally, improve on error handling for UserData class.

This specifically involves improving handling of invalid storage file directories.

antgyf committed:
test task package: create test cases for task classes

Create test suites for classes in the Johnson.task package.

antgyf committed:
Parser class: Implement more extensive testing

implement more test cases for parser class.

this will reduce the number of unforeseen bugs and errors.

antgyf committed:
create User Guide for Johnson

Create a User Guide that details the functionality of Johnson chatbot

antgyf committed:
Improve personality of chatbot

Improve upon personality of the chatbot by theming its responses.

Additionally improve JavaDocs throughout the codebase.

antgyf committed:
create AI.md file

Create AI.md file to document the usage of AI in the project.

antgyf committed:
Final submission

create necessary jar files and Ui.png for final submission
